code/modules/admin/watchlist.dm

```diff
@@ -0,0 +1,118 @@
+client/proc/watchlist_add(target_ckey, browse = 0)
+	if(!target_ckey)
+		var/new_ckey = ckey(input(usr,"Who would you like to add to the watchlist?","Enter a ckey",null) as text)
+		if(!new_ckey)
+			return
+		new_ckey = sanitizeSQL(new_ckey)
+		var/DBQuery/query_watchfind = dbcon.NewQuery("SELECT ckey FROM [format_table_name("player")] WHERE ckey = '[new_ckey]'")
+		if(!query_watchfind.Execute())
+			var/err = query_watchfind.ErrorMsg()
+			log_game("SQL ERROR obtaining ckey from player table. Error : \[[err]\]\n")
+			return
+		if(!query_watchfind.NextRow())
+			usr << "<span class='redtext'>[new_ckey] has not been seen before, you can only add known players.</span>"
```

Comment:
There are a few uses you could do with that, especially with noting down people from other servers. It is still arbitrary to limit a free-form text box to keys that have been here.